### DISCUSSION TITLE
Making circular frames

### DISCUSSION THREAD

**Discussion by srini8080** [ImageMagick/ImageMagick#6084] [Digital Image Processing]
IM 7+
Centos 8.3
I have a circularly cropped image to which I want to put a fancy frame

Can I get a Unix version of this script from Alan?

https://im.snibgo.com/fancyfr.htm#circ

Also if there is a way to emboss or bevel the frame only.

Thanks

**Comment by snibgo** [comment]
```
#! /bin/bash

ELEMENT=ffr_elem.png

declare $( magick \
  ${ELEMENT} \
  -format "EleW=%w\nEleH=%h\n" \
  info: )

nEle=10

PieceW=$(echo "${nEle}*${EleW}" | bc)

OutRad=%[fx:${PieceW}/2/Pi+${EleH}/2]
InRad=%[fx:${PieceW}/2/Pi-${EleH}/2]

magick \
  ${ELEMENT} -rotate 180 -write mpr:ELMT +delete \
  -size ${PieceW}x${EleH} xc: \
  -tile mpr:ELMT -draw "color 0,0 reset" \
  -virtual-pixel None \
  +distort polar ${OutRad},${InRad} \
  +repage \
  ffr_circ_frm.png


# Make a bevelled version:

magick.exe \
  ffr_circ_frm.png \
  -write mpr:INP +delete \
  \( mpr:INP \
     -shade 140x45 \
  \) \
  \( mpr:INP -colorspace Lab -separate \
  \) \
  -delete 1 \
  -combine \
  -set colorspace Lab -colorspace sRGB \
  ffr_bevelled.png
```

![ffr_bevelled](https://user-images.githubusercontent.com/33812368/219958385-e49dd71b-f538-423b-8bc8-11fb219fd220.png)

On embossing, see https://imagemagick.org/script/command-line-options.php#emboss and https://www.imagemagick.org/Usage/transform/#emboss

**Reply by srini8080** [reply]
Awesome,

Thanks much
